onnx/test/shape_inference_test.py

```diff
@@ -173,6 +175,46 @@ def _compare_value_infos(
                 "Unrecognized value info type in _compare_value_infos: ", str(vi_type)
             )
 
+    def _get_op_version(self, op_name):
+        """This function returns a version str list for an op.
+        If op has only one version, [] is returned.
+        If op has many version, [versions] is returned.
+        If op is not in the opset, None is returned.
+        """
+        op_names = dir(op_list)
+        if op_name in op_names:
+            return []
+
+        ret = []
+        name_pattern = re.compile(f"^{op_name}_([0-9]+)$")
+        for name in op_names:
+            m = name_pattern.search(name)
+            if m is not None:
+                ret.append(m.group(1))
+        if op_name in op_names or len(ret) > 0:
+            return ret
+        else:
+            return None
```

Comment:
It seems more direct to use `onnx.defs` to get all available versions for operators? For instance:
```
from onnx import defs
from onnx.defs import SchemaError
def get_available_versions(schema):
  versions = set()
  for version in range(schema.since_version, 0, -1):
   try:
     versions.add(defs.get_schema(schema.name, version).since_version)
   except SchemaError:
     break
  return versions
  
  
op_versions = {}
for schema in defs.get_all_schemas():
  if schema.domain == "":
    op_versions[schema.name] = get_available_versions(schema)
print(op_versions)
```

Reply:
That looks much better. I will try it later~

Reply:
This function works fine, so I used it in this PR. Thanks!